sdk/rust/oak/src/http/util.rs

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::{
+    io::{Receiver, ReceiverExt, Sender, SenderExt},
+    proto::oak::invocation::HttpInvocation,
+    OakError,
+};
+use log::error;
+use oak_abi::label::Label;
+use oak_services::proto::oak::encap::{HttpRequest, HttpResponse};
+
+/// A pair of temporary channels to pass the HTTP request to the Oak Node and receive the response.
+pub struct Pipe {
+    pub request_writer: Sender<HttpRequest>,
+    pub request_reader: Receiver<HttpRequest>,
+    pub response_writer: Sender<HttpResponse>,
+    pub response_reader: Receiver<HttpResponse>,
+}
+
+impl Pipe {
```

Comment:
Personally I'm not convinced this type makes things any simpler, it seems to just couple the API of the underlying senders and receivers.